mlir/lib/Dialect/Mesh/IR/MeshOps.cpp

```diff
@@ -129,6 +207,327 @@ MeshShardingAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// collective communication ops
+//===----------------------------------------------------------------------===//
+
+namespace {
```

Comment:
We don't use anonymous namespace outside of classes, instead we mark function `static`

Reply:
What is the reason for that? I think you can get linking collisions with static by violating the one-definition-rule.

Reply:
I was wrong about possible collisions.

Reply:
I made the functions static.